engine/lite/embeddedDeploymentManager/src/main/scala/pl/touk/nussknacker/engine/embedded/RequestResponseAkkaHttpHandler.scala

```diff
@@ -0,0 +1,98 @@
+        processInterpreters.get(processPath) match {
+          case None =>
+            complete {
+              HttpResponse(status = StatusCodes.NotFound)
+            }
+          case Some(processInterpreter) => new RequestResponseAkkaHttpHandler(processInterpreter).invoke {
+            case Invalid(errors) => complete {
+              logErrors(processPath, errors)
+              HttpResponse(status = StatusCodes.InternalServerError, entity = toEntity(errors.toList.map(info => EspError(info.nodeComponentInfo.map(_.nodeId), Option(info.throwable.getMessage)))))
+            }
+            case Valid(results) => complete {
+              HttpResponse(status = StatusCodes.OK, entity = toEntity(results))
+            }
+          }
+        }
+      }
+      //TODO place openApi endpoint
+    } ~ pathEndOrSingleSlash {
+      //healthcheck endpoint
+      get {
+        complete {
+          HttpResponse(status = StatusCodes.OK)
+        }
+      }
+    }
+
+  private def toEntity[T: Encoder](value: T): ResponseEntity = HttpEntity(contentType = `application/json`, string = value.asJson.noSpacesSortKeys)
+
+  private def logErrors(processPath: String, errors: NonEmptyList[NuExceptionInfo[_ <: Throwable]]): Unit = {
+    logger.warn(s"Failed to invoke: $processPath with errors: ${errors.map(_.throwable.getMessage)}")
+    errors.toList.foreach { error =>
+      logger.info(s"Invocation failed $processPath, error in ${error.nodeComponentInfo.map(_.nodeId)}: ${error.throwable.getMessage}", error.throwable)
+    }
+  }
+
+  @JsonCodec case class EspError(nodeId: Option[String], message: Option[String])
+
+}
+
+//this class handles parsing, displaying and invoking interpreter. This is the only place we interact with model, hence
+//only here we care about context classloaders
+class RequestResponseAkkaHttpHandler(requestResponseInterpreter: InterpreterType) extends Directives {
+
+  val invoke: Directive1[RequestResponseResultType[Json]] =
+    extractExecutionContext.flatMap { implicit ec =>
+      extractInput
+        .map(invokeInterpreter)
+        .flatMap(onSuccess(_))
+    }
+  private val source = requestResponseInterpreter.source
+  private val encoder = source.responseEncoder.getOrElse(DefaultResponseEncoder)
+  private val invocationMetrics = new InvocationMetrics(requestResponseInterpreter.context)
+  private val extractInput: Directive1[Any] = source match {
+    case a: RequestResponsePostSource[Any] =>
+      post & entity(as[Array[Byte]]).map(a.parse)
+    case a: RequestResponseGetSource[Any] =>
+      get & parameterMultiMap.map(a.parse)
+  }
+
+  private def invokeInterpreter(input: Any)(implicit ec: ExecutionContext): Future[RequestResponseResultType[Json]] = invocationMetrics.measureTime {
+    requestResponseInterpreter.invokeToOutput(input).map(_.andThen { data =>
+      Validated.fromTry(Try(encoder.toJsonResponse(input, data))).leftMap(ex => NonEmptyList.one(NuExceptionInfo(None, ex, Context(""))))
```

Comment:
why this logic is not in sink? why input is needed to encode output? The order looks like it is returned input as response :)

Reply:
By this I was thinking about such case that sink will produce `Json` and we will be sure that interpreter will handle this `leftMap(ex => NonEmptyList.one(NuExceptionInfo(None, ex, Context(""))))` correctly - will pass correct data, report metrics and so on. So in this handler will be need only handle data by casting it to `List[Json]`. WDYT? I'm not forcing it, we can think about it later

Reply:
added TODO for now